src/cards/soul_bind.ts

```diff
@@ -44,28 +44,33 @@ const spell: Spell = {
     add,
   },
   events: {
-    onTakeDamage: (unit, amount, underworld, prediction) => {
-      // Split the damage / healing amount among all soul bound units
-      const units = getSoulBoundUnits(underworld, prediction);
-      for (let i = 0; i < units.length; i++) {
-        const unit = units[i];
-        if (unit) {
-          const index = unit.events.findIndex(e => e == soulBindId);
-          // Remove Soul Bind damage event to prevent infinite loop
-          unit.events.splice(index, 1);
-          // Deal damage to unit
+    onTakeDamage: (unit, amount, underworld, prediction, damageDealer) => {
+      // Soul Bind splits damage/healing across all soul bound units
+      const boundUnits = getSoulBoundUnits(underworld, prediction);
+      for (let i = 0; i < boundUnits.length; i++) {
+        const boundUnit = boundUnits[i];
+        // This unit is already running the take damage event, so exclude it
+        if (boundUnit && boundUnit != unit) {
+          // Exception: Temporarily remove the bound unit's Soul Bind event to prevent infinite loop
+          const index = boundUnit.events.findIndex(e => e == soulBindId);
+          boundUnit.events.splice(index, 1);
```

Comment:
Another way to do this is
`boundUnit.events = boundUnit.events.filter(x => x !== souldBindId)`;
I like this method because you don't have to mess with indexes.  Note you must assign it back to events on the left side of the `=`

Reply:
Lol, I totally should've thought of that. I'll update this for both soul bind and thorns